src/main/java/com/powsybl/openloadflow/ac/IncrementalShuntVoltageControlOuterLoop.java

```diff
@@ -0,0 +1,194 @@
+        if (controllerContext.getDirectionChangeCount().getValue() <= MAX_DIRECTION_CHANGE) {
+            if (!controllerContext.getAllowedDirection().equals(direction.getAllowedDirection())) {
+                // both vs increase or decrease
+                // increase vs decrease or decrease vs increase
+                controllerContext.getDirectionChangeCount().increment();
+            }
+            controllerContext.setAllowedDirection(direction.getAllowedDirection());
+        }
+    }
+
+    private void adjustB(List<LfShunt> sortedControllerShunts, LfBus controlledBus, ContextData contextData, int[] controllerShuntIndex,
+                                              DenseMatrix sensitivities, double diffV, MutableObject<OuterLoopStatus> status) {
+        // several shunts control the same bus
+        double remainingDiffV = diffV;
+        boolean hasChanged = true;
+        while (hasChanged) {
+            hasChanged = false;
+            for (LfShunt controllerShunt : sortedControllerShunts) {
+                double sensitivity = ((EquationTerm<AcVariableType, AcEquationType>) controlledBus.getCalculatedV())
+                        .calculateSensi(sensitivities, controllerShuntIndex[controllerShunt.getNum()]);
+                // FIX ME: Not safe casting
+                LfShuntImpl controllerShuntImpl = (LfShuntImpl) controllerShunt;
+                double targetDeadband = controllerShuntImpl.getShuntVoltageControlTargetDeadband().orElse(null);
+                // Not very efficient because sorting is performed at each iteration. However, in practical should not be an issue.
+                // Considering storing the controlers sorted already as same order is used everywhere else
+                for (LfShuntImpl.Controller controller : controllerShuntImpl.getControllers().stream().sorted(Comparator.comparing(LfShuntImpl.Controller::getBMagnitude)).collect(Collectors.toList())) {
+                    var controllerContext = contextData.getControllersContexts().get(controller.getId());
+                    if (checkTargetDeadband(targetDeadband, remainingDiffV)) {
+                        double previousB = controller.getB();
+                        double deltaB = remainingDiffV / sensitivity;
+                        LfShuntImpl.Controller.Direction direction = controller.updateTapPositionB(deltaB, 1, controllerContext.getAllowedDirection()).orElse(null);
+                        if (direction != null) {
+                            updateAllowedDirection(controllerContext, direction);
+                            remainingDiffV -= (controller.getB() - previousB) * sensitivity;
+                            hasChanged = true;
+                            status.setValue(OuterLoopStatus.UNSTABLE);
+                            LOGGER.debug("Increment shunt susceptance of '{}': {} -> {}", controller.getId(), previousB, controller.getB());
```

Comment:
Susceptence value

Reply:
Just to check, this comment means that it is more proper to write  "Increment shunt susceptance value of" in the log?

Reply:
yes